### PR TITLE
fix(content-logic): enables separation of copy from business logic

### LINKED ISSUE #79: Allow separation of content from logic
Allow consumers to separate the content from the logic.

### PR DESCRIPTION
**Pull request by froehlichcr**
fixes #79

- aligns property names (e.g. drops `mode` in favor of `type`)
- further genericizes `IRef`
- in sample data, creates separate structures for content vs logic

This does not attempt to solve the separation of content in a generic way; rather, it focuses on demonstration that the separation is possible using the story data. Hopefully this will prove useful in creating a generic content vs logic loader in the future. For now, it meets the immediate need to allow for the separation of the two.

Rather than attempt to merge the entire JSON object of content with the entire object of logic, I opted to merge each known entity's content with the corresponding logic. There is still some risk that bad data could corrupt the survey, but that should happen inside the data construction.

### CODE REVIEW

File: src/schema/survey.ts [hunk 1]
```diff
@@ -75,34 +72,38 @@ export const survey = {
...
           "description": "Show an outline",
           "title": "Outline",
           "type": "boolean"
         },
+        "title": {
+          "title": "Title",
+          "type": "string"
+        },
+        "type": {
+          "$ref": "#/definitions/TButtonMode",
```
Comment (TomNUSDS) [inline src/schema/survey.ts:95]: Curious why you're using the `$` prefix for `ref` here?

File: src/stories/flows/eligibility/content/actions.flow.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/*
+  eslint-disable max-len,
+                sonarjs/no-duplicate-string,
+ */
+import { IButton }    from '../../../../survey';
+import { TActionMap } from '../lib/contentMap';
+
+const online: Partial<IButton> = {
+  id:    'ab1',
+  link:  'https://www.ssa.gov',
```
Comment (TomNUSDS) [inline src/stories/flows/eligibility/content/actions.flow.ts:10]: Did you want to keep ssa references out of the open source project?

Same goes for using phone numbers below.

The `example.com` domain is always safe to use. Also any phone number with 8675309 is blocklisted.

**Review by TomNUSDS — CHANGES_REQUESTED** [review]
You added a bunch of SSA specific flow data to the project. Can you confirm this is intentional before merging?

**Review by TomNUSDS — APPROVED** [review]
As per our chat. LGTM.

File: src/schema/survey.ts (hunk 1 above)
Comment (froehlichcr) [reply src/schema/survey.ts:95]: This file is auto-generated. This is the JSON schema derived from the interfaces.

**Pull request merged**